users/tests.py

```diff
@@ -1,3 +1,157 @@
+            first_name=first_name, last_name=last_name, email=email,
+            phone_number=phone_number, city=city, meeting_method=meeting_method,
+        )
+        return profile
+
+    return make
+
+
+@pytest.fixture
+def new_profile():
+    return Profile.create(username=USERNAME, password=PASSWORD, account_type=Account_type.STUDENT,
+                          first_name=FIRSTNAME, last_name=LASTNAME)
+
+
+@pytest.mark.django_db()
+class TestUserProfileModel():
+
+    def test_profile(self, new_profile):
+        assert new_profile.user.username == USERNAME
+        assert new_profile.user.first_name == FIRSTNAME
+
+    def test_profile_creation(self, new_profile):
+        new_profile.save()
+
+    def test_profile_invalid(self, new_profile):
+        assert new_profile
+
+    # Test for invalid profession length
+    def test_invalid_profession_length(self, new_profile):
+        with pytest.raises(ValidationError):
+            new_profile.save()
+            new_profile.profession = LONG_PROFESSION
+            new_profile.clean_fields()
+
+    # Test for invalid city
+    def test_invalid_city_length(self, new_profile):
+        with pytest.raises(ValidationError):
+            new_profile.save()
+            new_profile.city = LONG_PROFESSION
+            new_profile.clean_fields()
+
+
+@pytest.mark.django_db
+class TestProfileModel:
+    def test_new_profile(self, make_profile):
+        profile = make_profile()
+        assert profile.account_type == PROFIL_INFORMATION.get('account_type')[0]
+        assert profile.phone_number == PROFIL_INFORMATION.get('phone_number')[0]
+        assert profile.city == PROFIL_INFORMATION.get('city')[0]
+        assert profile.meeting_method == PROFIL_INFORMATION.get('meeting_method')[0]
+
+    def test_get_profile(self, make_profile):
+        profile = make_profile()
+        assert profile in Profile.objects.all()
+
+    def test_filter_by_city(self, make_profile):
+        new_profile = make_profile(username=PROFIL_INFORMATION.get('username')[1],
+                                   password=PROFIL_INFORMATION.get('password')[1],
+                                   email=PROFIL_INFORMATION.get('email')[1],
+                                   phone_number=PROFIL_INFORMATION.get('phone_number')[1])
+        make_profile(username=PROFIL_INFORMATION.get('username')[2],
+                     password=PROFIL_INFORMATION.get('password')[2],
+                     city='Toronto',
+                     email='john2.doe@example.com',
+                     phone_number='0541234636')
+        make_profile(username=PROFIL_INFORMATION.get('username')[0],
+                     password=PROFIL_INFORMATION.get('password')[0],
+                     city='London',
+                     email='john3.doe@example.com',
+                     phone_number='0541234563')
+
+        assert list(Profile.filter_by_city(PROFIL_INFORMATION.get('city')[0])) == [new_profile]
```

Comment:
why do we create multiple profiles here?

Reply:
I wanted to check a couple of different options to filter them(I don't have test data)